### PR TITLE
Switch persistence to microstream.one

### PR DESCRIPTION
**Pull request by cc-cpo**

### CODE REVIEW

File: src/main/java/de/wirvsvirus/hack/repository/OnboardingRepositoryMicrostream.java [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+
+  @Override
+  public User lookupUserById(final UUID userId) {
+    Preconditions.checkNotNull(userId);
+
+    return
+        EntryStream.of(database.dataRoot().getAllUsers())
+            .values()
+            .collect(MoreCollectors.onlyOne(user -> user.getUserId().equals(userId)))
+            .orElseThrow(() -> new IllegalStateException("User not found by id " + userId));
```
Comment (BendingBender) [inline src/main/java/de/wirvsvirus/hack/repository/OnboardingRepositoryMicrostream.java]:
```suggestion
            .orElseThrow(() -> new IllegalStateException("Unique user not found by id " + userId));
```

File: src/main/java/de/wirvsvirus/hack/repository/OnboardingRepositoryMicrostream.java [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+    message.setCreatedAt(Instant.now());
+    message.setSenderUserId(sender.getUserId());
+    message.setRecipientUserId(recipient.getUserId());
+    message.setText(text);
+
+    messages.add(message);
+
+    database.persist(messages);
+    // TODO try this instead
+//    database.persist(database.reloadRoot().getAllGroupMessages());
```
Comment (BendingBender) [inline src/main/java/de/wirvsvirus/hack/repository/OnboardingRepositoryMicrostream.java]: Wut?

**Review by BendingBender — APPROVED** [review]

File: src/main/java/de/wirvsvirus/hack/repository/OnboardingRepositoryMicrostream.java (hunk 2 above)
Comment (cc-cpo) [reply src/main/java/de/wirvsvirus/hack/repository/OnboardingRepositoryMicrostream.java]: done

File: src/main/java/de/wirvsvirus/hack/repository/OnboardingRepositoryMicrostream.java (hunk 1 above)
Comment (cc-cpo) [reply src/main/java/de/wirvsvirus/hack/repository/OnboardingRepositoryMicrostream.java]: done

**Review by BendingBender — APPROVED** [review]

**Pull request merged**